build_tools/therock_build_observability.py

```diff
@@ -0,0 +1,390 @@
+#!/usr/bin/env python3
+"""
+author: shiraz.ali@amd.com
+
+Tooling for TheRock build system. The goal is to have single unified view across build resource utilization for therock compoments and systems locally as well as in CI
+
+Features:
+- Acts as CMake compiler launcher (C/C++).
+- Logs per-command timing + memory usage to /tmp/therock-build-resources (or THEROCK_BUILD_PROF_LOG_DIR).
+- Aggregates all logs into:
+    - comp-summary.csv  (per-component summary)
+    - comp-summary.md   (Markdown table per-component)
+- NEVER fails the build if profiling/reporting fails.
+  
+- Only propagates the real compiler's exit code.
+
+Usage in CMake configure:
+
+  cmake -S . -B build -GNinja \
+    -DTHEROCK_AMDGPU_FAMILIES=gfx110X-all \
+    -DCMAKE_C_COMPILER_LAUNCHER="${PWD}/build_tools/therock_build_observability.py" \
+    -DCMAKE_CXX_COMPILER_LAUNCHER="${PWD}/build_tools/therock_build_observability.py"
+
+
+  cmake --build build 
+          OR 
+  ninja -C build -j"$(nproc)"
+
+
+Output: 
+    End summary for observability of build resources gets created in /tmp/comp-summary.md /tmp/comp-summary.csv file 
+
+
+"""
+
+import os
+import sys
+import time
+import random
+import datetime
+import resource
+import subprocess
+import shlex
+from typing import Dict, Tuple
+
+
+# -------------------------
+# Component classification
+# -------------------------
+
+def guess_component_from_pwd_and_cmd(pwd: str, cmd_str: str) -> str:
+    """ when not clear to infer which component this compile belongs to."""
+    comp = "unknown"
+
+    # Infer from build directory (PWD). Can be edited, added as needed
+    if "/core/" in pwd:
+        comp = "core"
+    elif "/compiler/" in pwd:
+        comp = "compiler"
```

Comment:
I don't think we should do any "guessing" here. The build system knows what directories it is working in and what the CMake subprojects are named.

We should be able to match the existing log files in `build/logs/` with names like
* `hipBLAS_configure.log`
* `hipBLAS_build.log`
* `hipBLAS_install.log`

How about generating new files next to those like
* `hipBLAS_build_stats.log`
* `hipBLAS_build.stats_log` (similar to `.ninja_log` files)

?